lib/src/file_context.dart

```diff
@@ -37,29 +37,34 @@ class FileContext {
   /// text, which is useful for the creation of [SourcePatch]es.
   SourceFile get sourceFile =>
       _sourceFile ??= SourceFile.fromString(sourceText, url: Uri.file(path));
-  SourceFile _sourceFile;
+  SourceFile? _sourceFile;
```

Comment:
#nit This private variable and getter could be replaced with a `late` variable

https://dart.dev/null-safety/understanding-null-safety#lazy-initialization
```dart
final late SourceFile sourceFile = SourceFile.fromString(sourceText, url: Uri.file(path);
```

Same with `sourceText`/`_contents`

Reply:
Woah! I had no idea you could do this, that's awesome!